File: 2012/submissions/Khalil Fazal/1/C++/Main.cpp
#include <iostream>
#include <string>
using namespace std;

typedef unsigned int uint;

int firstIndex(const string word, const string input) {
    for (uint i = 0; i < input.length(); i++) {
        uint j;

        for (j = 0; j < word.length() && i + j < input.length() && input[i + j] == word[j]; j++);

        if (j == word.length()) {
            return i;
        }
    }

    // Word doesn't exit in input
    return -1;
}

int main(const int argc, const char** argv) {
    if (argc > 1) {
        string input;

        while (getline(cin, input)) {
            cout << firstIndex(argv[1], input) << endl;
        }
    }

    return 0;
}

File: 2012/submissions/Khalil Fazal/1/Java/Main.java
import java.util.Scanner;

public class Main {
    private static int firstIndex(final String word, final String input) {
        for (int i = 0; i < input.length(); i++) {
            int j;

            for (j = 0; j < word.length() && i + j < input.length() && input.charAt(i + j) == word.charAt(j); j++);

            if (j == word.length()) {
                return i;
            }
        }

        // Word doesn't exit in input
        return -1;
    }

    public static void main(final String[] args) {
        if (args.length > 0) {
            final Scanner in = new Scanner(System.in);

            while (in.hasNextLine()) {
                System.out.println(firstIndex(args[0], in.nextLine()));
            }

            in.close();
        }
    }
}
